srcs/execution/exec.c

```diff
@@ -0,0 +1,98 @@
+#include "libft.h"
+#include "shell_env.h"
+#include "env_tools.h"
+#include "exec.h"
+#include <unistd.h>
+
+/*
+** Retrieve env vars and PATH
+*/
+
+void	pre_exec(char **cmd)
+{
+	char	**path;
+	char	**env;
+ 
+	env = tabify(get_shell_env()->variables);
+	path = ft_strsplit(get_tenv(env, "PATH"), ':');
+	execute(cmd, env, path);
+	ft_freetabchar(path);
+	ft_freetabchar(env);
+}
+
+/*
+** If local executable:
+** Check file exist.
+** Check user have executable rights on it.
+** Execute it and exit fork.
+** Else: execute path
+** exit the fork if the no execution because execution stop the fork
+*/
+
+void	execute(char **cmd, char **env, char **path)
+{
+	if (cmd[0][0] == '.' && cmd[0][1] == '/' && cmd[0][2] != '\0')
```

Comment:
man execvp: The functions execlp(), execvp(), and execvP() will duplicate the actions of the shell in searching for an executable file if the specified file name does not contain a slash '/' character.
pour /bin/ls, etc. aussi